fvtest/porttest/si.cpp

```diff
@@ -2362,6 +2368,289 @@ TEST(PortSysinfoTest, sysinfo_test_os_kernel_info)
 	return;
 }
 
+class CgroupTest : public ::testing::Test {
+protected:
+#if defined(LINUX)
+	static bool isV1Available;
+	static bool isV2Available;
+	static std::string memCgroup;
+	static std::string cpuCgroup;
+	const static std::map<std::string, uint64_t> supportedSubsystems;
+	constexpr static char CGROUP_MOUNT_POINT[] = "/sys/fs/cgroup";
+	static uint64_t available;
+	static bool setUpSucceeded;
+
+	/* Initialize static vars. */
+	static void
+	SetUpTestCase()
+	{
+		isV1Available = isCgroupV1Available();
+		isV2Available = isCgroupV2Available();
+		ASSERT_TRUE(isV1Available != isV2Available);
+
+		std::ifstream cgroupFile(std::string("/proc/") + std::to_string(getpid()) + std::string("/cgroup"));
+		std::string line;
+		if (isV1Available) {
+			while (std::getline(cgroupFile, line)) {
```

Comment:
`getline` returns `istream` not a boolean.
```suggestion
			while (NULL != std::getline(cgroupFile, line)) {
```

Reply:
All examples use the return value as a condition. Also I can't compare istream to null
```
/root/docker_hostdir/temp/omr/fvtest/porttest/si.cpp:2394:16: error: no match for ‘operator!=’ (operand types are ‘long int’ and ‘std::basic_istream<char>’)
    while (NULL != std::getline(cgroupFile, line)) {
```
istream inherits operator bool https://en.cppreference.com/w/cpp/io/basic_ios/operator_bool so that it can be used in conditions. I think I will add an explicit cast `(bool)std::getline(cgroupFile, line)`